testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/security/SslServerWithWildcardHostnameCertificateTest.java

```diff
@@ -0,0 +1,121 @@
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import javax.ws.rs.ProcessingException;
+import javax.ws.rs.core.Response;
+
+import static org.jboss.resteasy.test.ContainerConstants.SSL_CONTAINER_PORT_OFFSET_WILDCARD;
+import static org.jboss.resteasy.test.ContainerConstants.SSL_CONTAINER_QUALIFIER_WILDCARD;
+
+/**
+ * @tpSubChapter Security
+ * @tpChapter Integration tests
+ * @tpTestCaseDetails Tests for SSL - server secured with certificate with wildcard hostname "*host"
+ * @tpSince RESTEasy 3.7.0
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+public class SslServerWithWildcardHostnameCertificateTest extends SslTestBase {
+
+   private static final Logger LOG = Logger.getLogger(SslServerWithWildcardHostnameCertificateTest.class.getName());
+
+   private static KeyStore truststore;
+
+   private static final String SERVER_KEYSTORE_PATH = RESOURCES + "/server-wildcard-hostname.keystore";
+   private static final String CLIENT_TRUSTSTORE_PATH = RESOURCES + "/client-wildcard-hostname.truststore";
+   private static final String URL = generateHttpsURL(SSL_CONTAINER_PORT_OFFSET_WILDCARD);
+
+   @TargetsContainer(SSL_CONTAINER_QUALIFIER_WILDCARD)
+   @Deployment(managed=false, name=DEPLOYMENT_NAME)
+   public static Archive<?> createDeployment() {
+      WebArchive war = TestUtil.prepareArchive(DEPLOYMENT_NAME);
+      return TestUtil.finishContainerPrepare(war, null, SslResource.class);
+   }
+
+   @BeforeClass
+   public static void prepareTruststore() throws KeyStoreException, IOException, CertificateException, NoSuchAlgorithmException {
+      truststore = KeyStore.getInstance("jks");
+      try (InputStream in = new FileInputStream(CLIENT_TRUSTSTORE_PATH)) {
+         truststore.load(in, PASSWORD.toCharArray());
+      }
+   }
+
+   @Before
+   public void startContainer() throws Exception {
+      if (!containerController.isStarted(SSL_CONTAINER_QUALIFIER_WILDCARD)) {
+         containerController.start(SSL_CONTAINER_QUALIFIER_WILDCARD);
+         secureServer(SERVER_KEYSTORE_PATH, SSL_CONTAINER_PORT_OFFSET_WILDCARD);
+         deployer.deploy(DEPLOYMENT_NAME);
+      }
+   }
+
+   /**
+    * @tpTestDetails HostnameVerificationPolicy.WILDCARD test
+    * Client has truststore containing self-signed certificate.
+    * Server/endpoint is secured with the same self-signed certificate, but only wildcard of server hostname (*host) is included among 'subject alternative names' in the certificate.
+    * Client should trust the server because HostnameVerificationPolicy is set to WILDCARD.
+    * @tpSince RESTEasy 3.7.0
+    */
+   @Test
+   public void testHostnameVerificationPolicyWildcard() {
+      resteasyClientBuilder = new ResteasyClientBuilderImpl();
```

Comment:
@tterem @marekkopecky , I'm not going to block the merging of the PR because of this, but we should now try avoiding creating the ResteasyClientBuilder like this in tests, as users might think that's a best practice. We should create the builder using teh JAX-RS API and cast it to the ResteasyClientBuilder interface, as the ResteasyClientBuilderImpl is internal implementation now. Just for future reference :-)